dish_grpc_influx2.py

```diff
@@ -0,0 +1,330 @@
+#!/usr/bin/python3
+"""Write Starlink user terminal data to an InfluxDB 2.x database.
+
+This script pulls the current status info and/or metrics computed from the
+history data and writes them to the specified InfluxDB 2.x database either once
+or in a periodic loop.
+
+Data will be written into the requested database with the following
+measurement / series names:
+
+: spacex.starlink.user_terminal.status : Current status data
+: spacex.starlink.user_terminal.history : Bulk history data
+: spacex.starlink.user_terminal.ping_stats : Ping history statistics
+: spacex.starlink.user_terminal.usage : Usage history statistics
+
+NOTE: The Starlink user terminal does not include time values with its
+history or status data, so this script uses current system time to compute
+the timestamps it sends to InfluxDB. It is recommended to run this script on
+a host that has its system clock synced via NTP. Otherwise, the timestamps
+may get out of sync with real time.
+"""
+
+from datetime import datetime
+from datetime import timezone
+import logging
+import os
+import signal
+import sys
+import time
+import warnings
+
+from influxdb_client import InfluxDBClient, WriteOptions, WritePrecision
+
+import dish_common
+
+HOST_URL = "localhost:8086"
+BUCKET_DEFAULT = "starlinkstats"
+BULK_MEASUREMENT = "spacex.starlink.user_terminal.history"
+FLUSH_LIMIT = 6
+MAX_BATCH = 5000
+MAX_QUEUE_LENGTH = 864000
+
+
+class Terminated(Exception):
+    pass
+
+
+def handle_sigterm(signum, frame):
+    # Turn SIGTERM into an exception so main loop can clean up
+    raise Terminated
+
+
+def parse_args():
+    parser = dish_common.create_arg_parser(output_description="write it to an InfluxDB 2.0 database")
+
+    group = parser.add_argument_group(title="InfluxDB 2.0 database options")
+    group.add_argument("-u",
+                       "--url",
+                       default=HOST_URL,
+                       dest="url",
+                       help="URL or IP AND Port of the InfluxDB 2.0 Host, default: " + HOST_URL)
+    group.add_argument("-T", "--token", help="Token to access the bucket")
+    group.add_argument("-B",
+                       "--bucket",
+                       default=BUCKET_DEFAULT,
+                       help="Bucket name to use, default: " + BUCKET_DEFAULT)
+    group.add_argument("-O", "--org", help="Organisation name")
+    group.add_argument("-k",
+                       "--skip-query",
+                       action="store_true",
+                       help="Skip querying for prior sample write point in bulk mode")
+    group.add_argument("-C",
+                       "--ca-cert",
+                       dest="verify_ssl",
```

Comment:
This option did not work for me. It looks like the API changed so that CA cert file is no longer overloaded with "verify_ssl" the way it was in the older InfluxDB client and that there is now a separate "ssl_ca_cert" kwarg.